File: internacao/src/main/java/br/com/basis/madre/service/dto/SolicitacaoDeInternacaoDTO.java
package br.com.basis.madre.service.dto;

import br.com.basis.madre.domain.enumeration.Prioridade;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class SolicitacaoDeInternacaoDTO implements Serializable {

    private Long id;

    @NotNull
    private Long pacienteId;

    private String nomeDoPaciente;

    @NotNull
    private LocalDate dataProvavelDaInternacao;

    private LocalDate dataProvavelDaCirurgia;

    @NotNull
    private Prioridade prioridade;

    @NotNull
    private String principaisSinaisESintomasClinicos;

    @NotNull
    private String condicoesQueJustificamInternacao;

    @NotNull
    private String principaisResultadosProvasDiagnosticas;


    private Long cidPrincipalId;

    private Long cidSecundarioId;

    private Long equipeId;

    private Long crmId;

    private Long procedimentoId;

}

File: frontend/src/app/internacao/models/dtos/solicitacao-de-internacao.dto.ts
export interface ISolicitacaoDeInternacaoDTO {
    id?: number;
    pacienteId?: number;
    dataProvavelDaInternacao: Date;
    dataProvavelDaCirurgia: Date;
    prioridade: string;
    principaisSinaisESintomasClinicos: string;
    condicoesQueJustificamInternacao: string;
    principaisResultadosProvasDiagnosticas: string;
    cidPrincipalId: number;
    cidSecundarioId: number;
    equipeId: number;
    crmId: number;
    procedimentoId: number;
}

export class SolicitacaoDeInternacaoDTO implements ISolicitacaoDeInternacaoDTO {
    public id?: number;
    public pacienteId?: number;
    public dataProvavelDaInternacao: Date;
    public dataProvavelDaCirurgia: Date;
    public prioridade: string;
    public principaisSinaisESintomasClinicos: string;
    public condicoesQueJustificamInternacao: string;
    public principaisResultadosProvasDiagnosticas: string;
    public cidPrincipalId: number;
    public cidSecundarioId: number;
    public equipeId: number;
    public crmId: number;
    public procedimentoId: number;
}
